sled-agent/src/config.rs

```diff
@@ -15,14 +15,15 @@ use illumos_utils::dladm::PhysicalLink;
 use omicron_common::vlan::VlanID;
 use serde::Deserialize;
 use sled_hardware::UnparsedDisk;
-use sled_hardware::is_gimlet;
+use sled_hardware::is_oxide_sled;
 use sprockets_tls::keys::SprocketsConfig;
 
 #[derive(Clone, Debug, Deserialize)]
 #[serde(rename_all = "lowercase")]
 pub enum SledMode {
     Auto,
-    Gimlet,
+    #[serde(alias = "gimlet")]
+    Sled,
```

Comment:
Super nitpicky, feel free to ignore me: the name "sled" for non-scrimlet sleds feels a bit iffy to me, since a scrimlet is also a sled...but I can't immediately think of a better alternative. I considered suggesting "compute" for a sled that doesn't have switch-attached responsibilities, but don't really love that since (a) they also provide storage and (b) scrimlets can also run VMs. Something like `NonSwitchAttached` feels like an awful mouthful, and I don't know how  I feel about something like `Normal`. The more I think about it, "sled" is probably the least unpleasant name. 🤷‍♀️

Reply:
I, for one, am strongly in favor of preserving "scrimlet" for future generations.